### DISCUSSION TITLE
Getting Started with GitHub Part 1: Best practices for setting up your GitHub account

### DISCUSSION THREAD

**Discussion by potaders** [community/community#39082] [Discover: GitHub Best Practices]
### Creating a username

Welcome to GitHub! Thank you for starting your journey on the largest developer community in the world!

The first thing you need to do when using GitHub is create an account. Whether you're working in open source projects, working with friends on a school project, or just exploring a new tool that you overheard some people talking about at work, a GitHub account grants you basic access to the platform.

To create your account, you'll need to have 3 things:

- A unique username, or handle :tada:
- An email address :mailbox:
- A password :lock:

### Your handle

At GitHub, your handle is super important. It is a public reference that can be traced back to all public (and sometimes private) things you do on GitHub. That can seem daunting, but it can also be fun! Handles at GitHub are an individual thing, no 2 are exactly the same, so take the time to decide on something that shows off the amazing person that you are. Selecting a handle can be a hard decision, thats where we can help!

#### Be yourself

As stated above, handles are highly visible, so it only makes sense that they should show who you are as individual or who you want to be perceived as. Please keep in mind that a handle is also something that you should feel comfortable being public. Handles are often added to resumes so having a handle that is respectful may help recruiters feel comfortable about bringing in a candidate for a job interview. 

Handles also area a great opportunity to show off more about yourself than your development. Maybe you have a love of a certain type of music, maybe you connect with a certain comic book character, or maybe you're an animal lover, etc. Keep in mind that trends sometimes pass, so creating a handle that is associated with a specific topic could age poorly.

Quick note: GitHub handles may only contain 39 alphanumeric characters and a hyphen (ex. `good-handle`). Hyphens can only be used once in a handle (ex. `not-allowed-handle`) and cannot be at the beginning or end of the handle (ex. `-notallowed-`). GitHub is a community for everyone, please consider reading [this page](https://docs.github.com/en/site-policy/acceptable-use-policies/github-hate-speech-and-discrimination) as well before creating a handle.

**Attention**: It is technically possible to change your GitHub handle, doing so will not update any references to your past handle. To learn more, review documentation found [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/changing-your-github-username).

At the end of the day, your GitHub handle will become your calling card when interacting with all things GitHub. Other GitHub users will use it as a reference when asking you questions, assigning you tasks, or looking at your public contributions. More info [here](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy).

### Creating a repository

Now that you have a handle, you can now create a repository. You can create repositories in a few places, but the easiest place to start is in your personal org. In the upper right corner of the GitHub.com, you will see a plus sign. Clicking on that will provide a prompt allowing you to create a new repository. 

Repo creation screen should look something like this: 
![repo-create-screenshot-1](https://user-images.githubusercontent.com/103224897/201693675-44049ae5-036d-40a2-b798-0337cc00aca3.png)

When creating a repository, the only required detail is the name of the repository. Repository names cannot have spaces, so any space you leave in your name will automatically be filled with dashes `-`. Optionally, you can include a description which can be used to help provide details about what is in the repository. For example, if you're working on a mobile app, you could name the repository "My-cool-app" and provide a description of "My mobile app written in React Native". This makes the repository much more approachable in the event that you want to share your cool app with the world!

#### Public or private repositories?

When creating a repository you have the option of setting the repository as public or private. Public repositories are exactly as they sound, public! They are open for the world to see and can be viewed by anyone on GitHub. This is a good route if you're looking to share your work and potentially get public feedback and contributions. In contrast to public, private repositories are not visible to anyone other than you. With private repositories you also have the power to invite in other GitHub users to view and collaborate on your code.

The default choice when creating a repository is private, though it's not always the best choice. Say you're working on a new React Native app and you make a really slick menu bar. If your repository is public, then anyone who finds it would be able to use your code and improve their apps. By keeping your project public, anyone who finds your project could help you make improvements to your code through Issues and Pull Requests.

#### Initialize with a README.md?

The final step to creating a repository is deciding whether to initiate it with a README. At GitHub, nearly all of our repositories have a README files and they serve a very important purpose. Creating your repository with a README is an excellent choice if you are starting from scratch and can help save time by avoiding the need to import an existing git project. 

For those who are planning to import an existing git project, it is best to avoid selecting the 'initiate with README' checkbox. This will help you avoid conflicts in your git history when pushing the contents of your existing project into your new GitHub repository.

As stated above, at GitHub, the README file serves a very important purpose. The README(.md) file is extra special because it will be automatically shown as the first thing you see when viewing your GitHub repository from a web browser. Think of the README file as a guide to all of the things in your repository; it's the place where you get to share all of the important information about your repository and show others how to use your code. Visitors and contributors to your repository should be able to read through your project's README file and understand what your project does, who to contact for questions, how they can use it, and how to contribute.

README files are written in language called Markdown, which supports showing many varieties of information including images, links, GIFs, emojis, tables, and many others! While Markdown cheatsheets can be found all over the internet, GitHub does have some helpful docs that can assist you in getting you started [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).

To paint a picture, let's take a look at a README file for a very popular open source tool on GitHub, [Visual Studio Code](https://github.com/microsoft/vscode). The first thing you may notice in the README are some helpful 'badges'. These badges can be used to show all sorts of helpful information, in this example, VS Code has chosen to show issue counts and open bugs. Other popular badges include showing information about build status from a CI (Continuous Integration) tool, chat ops connection like Slack, or test coverage status. Next, as we move down the VS Code README, we see more information about the product, how to make contributions, how to provide feedback, and additional information about licensing, building, and a code of conduct. Not all of this information is needed in your README, but it can be very helpful to get an idea about what you'd like to share about your project.

The README files are a great starting point to your project, but they are far from the final step. You should expect your README file to change as your project changes, so when your React app grows to include some new features, you should expect your README to grow too!

### Welcome to GitHub!

That's all for now; welcome to GitHub. We are overjoyed that you have joined the largest developer community in the world. From amazing handle, to fresh code, we are excited build with you from here, together!

**Update!** 
Links to the other parts of this of this series:
- [Part 2](https://github.com/orgs/community/discussions/47569)
- [Part 3](https://github.com/orgs/community/discussions/72264)

**Comment by Nassarjj1** [comment]
> ### Creating a username
> 
> 
> 
> Welcome to GitHub! Thank you for starting your journey on the largest developer community in the world!
> 
> 
> 
> The first thing you need to do when using GitHub is create an account. Whether you're working in open source projects, working with friends on a school project, or just exploring a new tool that you overheard some people talking about at work, a GitHub account grants you basic access to the platform.
> 
> 
> 
> To create your account, you'll need to have 3 things:
> 
> 
> 
> - A unique username, or handle :tada:
> 
> - An email address :mailbox:
> 
> - A password :lock:
> 
> 
> 
> ### Your handle
> 
> 
> 
> At GitHub, your handle is super important. It is a public reference that can be traced back to all public (and sometimes private) things you do on GitHub. That can seem daunting, but it can also be fun! Handles at GitHub are an individual thing, no 2 are exactly the same, so take the time to decide on something that shows off the amazing person that you are. Selecting a handle can be a hard decision, thats where we can help!
> 
> 
> 
> #### Be yourself
> 
> 
> 
> As stated above, handles are highly visible, so it only makes sense that they should show who you are as individual or who you want to be perceived as. Please keep in mind that a handle is also something that you should feel comfortable being public. Handles are often added to resumes so having a handle that is respectful may help recruiters feel comfortable about bringing in a candidate for a job interview. 
> 
> 
> 
> Handles also area a great opportunity to show off more about yourself than your development. Maybe you have a love of a certain type of music, maybe you connect with a certain comic book character, or maybe you're an animal lover, etc. Keep in mind that trends sometimes pass, so creating a handle that is associated with a specific topic could age poorly.
> 
> 
> 
> Quick note: GitHub handles may only contain 39 alphanumeric characters and a hyphen (ex. `good-handle`). Hyphens can only be used once in a handle (ex. `not-allowed-handle`) and cannot be at the beginning or end of the handle (ex. `-notallowed-`). GitHub is a community for everyone, please consider reading [this page](https://docs.github.com/en/site-policy/acceptable-use-policies/github-hate-speech-and-discrimination) as well before creating a handle.
> 
> 
> 
> **Attention**: It is technically possible to change your GitHub handle, doing so will not update any references to your past handle. To learn more, review documentation found [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/changing-your-github-username).
> 
> 
> 
> At the end of the day, your GitHub handle will become your calling card when interacting with all things GitHub. Other GitHub users will use it as a reference when asking you questions, assigning you tasks, or looking at your public contributions. More info [here](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy).
> 
> 
> 
> ### Creating a repository
> 
> 
> 
> Now that you have a handle, you can now create a repository. You can create repositories in a few places, but the easiest place to start is in your personal org. In the upper right corner of the GitHub.com, you will see a plus sign. Clicking on that will provide a prompt allowing you to create a new repository. 
> 
> 
> 
> Repo creation screen should look something like this: 
> 
> ![repo-create-screenshot-1](https://user-images.githubusercontent.com/103224897/201693675-44049ae5-036d-40a2-b798-0337cc00aca3.png)
> 
> 
> 
> 
> 
> When creating a repository, the only required detail is the name of the repository. Repository names cannot have spaces, so any space you leave in your name will automatically be filled with dashes `-`. Optionally, you can include a description which can be used to help provide details about what is in the repository. For example, if you're working on a mobile app, you could name the repository "My-cool-app" and provide a description of "My mobile app written in React Native". This makes the repository much more approachable in the event that you want to share your cool app with the world!
> 
> 
> 
> #### Public or private repositories?
> 
> 
> 
> When creating a repository you have the option of setting the repository as public or private. Public repositories are exactly as they sound, public! They are open for the world to see and can be viewed by anyone on GitHub. This is a good route if you're looking to share your work and potentially get public feedback and contributions. In contrast to public, private repositories are not visible to anyone other than you. With private repositories you also have the power to invite in other GitHub users to view and collaborate on your code.
> 
> 
> 
> The default choice when creating a repository is private, though it's not always the best choice. Say you're working on a new React Native app and you make a really slick menu bar. If your repository is public, then anyone who finds it would be able to use your code and improve their apps. By keeping your project public, anyone who finds your project could help you make improvements to your code through Issues and Pull Requests.
> 
> 
> 
> #### Initialize with a README.md?
> 
> 
> 
> The final step to creating a repository is deciding whether to initiate it with a README. At GitHub, nearly all of our repositories have a README files and they serve a very important purpose. Creating your repository with a README is an excellent choice if you are starting from scratch and can help save time by avoiding the need to import an existing git project. 
> 
> 
> 
> For those who are planning to import an existing git project, it is best to avoid selecting the 'initiate with README' checkbox. This will help you avoid conflicts in your git history when pushing the contents of your existing project into your new GitHub repository.
> 
> 
> 
> As stated above, at GitHub, the README file serves a very important purpose. The README(.md) file is extra special because it will be automatically shown as the first thing you see when viewing your GitHub repository from a web browser. Think of the README file as a guide to all of the things in your repository; it's the place where you get to share all of the important information about your repository and show others how to use your code. Visitors and contributors to your repository should be able to read through your project's README file and understand what your project does, who to contact for questions, how they can use it, and how to contribute.
> 
> 
> 
> README files are written in language called Markdown, which supports showing many varieties of information including images, links, GIFs, emojis, tables, and many others! While Markdown cheatsheets can be found all over the internet, GitHub does have some helpful docs that can assist you in getting you started [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).
> 
> 
> 
> To paint a picture, let's take a look at a README file for a very popular open source tool on GitHub, [Visual Studio Code](https://github.com/microsoft/vscode). The first thing you may notice in the README are some helpful 'badges'. These badges can be used to show all sorts of helpful information, in this example, VS Code has chosen to show issue counts and open bugs. Other popular badges include showing information about build status from a CI (Continuous Integration) tool, chat ops connection like Slack, or test coverage status. Next, as we move down the VS Code README, we see more information about the product, how to make contributions, how to provide feedback, and additional information about licensing, building, and a code of conduct. Not all of this information is needed in your README, but it can be very helpful to get an idea about what you'd like to share about your project.
> 
> 
> 
> The README files are a great starting point to your project, but they are far from the final step. You should expect your README file to change as your project changes, so when your React app grows to include some new features, you should expect your README to grow too!
> 
> 
> 
> ### Welcome to GitHub!
> 
> 
> 
> That's all for now; welcome to GitHub. We are overjoyed that you have joined the largest developer community in the world. From amazing handle, to fresh code, we are excited build with you from here, together!

**Comment by Edgdangon1** [comment]
hello I am new in gitHub, could somebody tell me how to delete a remote and local repository.
Thanks

**Reply by DSDmark** [reply]
Hi @potaders,

it's not good things to ask question here. Next time, if you want to ask, then create discussion.
Anyway,You can use this ```git remote remove origin``` or ```git remote rm origin```

**Reply by Edgdangon1** [reply]
ok Thanks

**Reply by tuves** [reply]
@Edgdangon1, you can navigate to the [Repositories](https://github.com/orgs/community/discussions/categories/repositories) category to ask questions regarding repositories. In our left-hand side menu, you'll see all the categories on which you can ask questions, provide feedback, and share knowledge.


![Screen Shot 2022-11-17 at 9 59 44 AM](https://user-images.githubusercontent.com/49915749/202416984-e0e96244-5d77-47f0-90b1-1caaf8900907.png)

I hope that's helpful--and welcome to the GitHub community! :)

**Comment by victorl1m** [comment]
it's a good guide to new users.. thx for this!

**Reply by esther-kpet** [reply]
Yes, found it helpful too

**Comment by Georgeododo** [comment]
I have enjoyed this message. I am happy at the amazing story. I am now very much eager to create my Github account which I have laboured for more than 24 hours.

**Reply by AVI-SHUKLA1110** [reply]
THANKS FOR THIS

**Comment by Jena123J** [comment]
Hi mi name jena`

**Comment by Electronic-Biker** [comment]
Hi guys, I'm new to GitHub. I am here because I have been reading the Micro:Python manual and I have spotted a serious error in some versions that could confuse many users of the micro:bit and the micro:python language. Being a new user of the micro:bit it certainly confused me!
In the book I was reading the inclinometer (sometimes called a 'tilt indicator') was referred to as an accelerometer, which it clearly isn't. An accelerometer measures the speed of movement of an object that it is attached to. If the speed stays constant then acceleration is zero. The micro:bit does not do that, it measures angles with respect to the direction of gravity. You could simulate it with a protractor and a plumb line.
I eventually found an almost identical manual that used the correct terminology, but it was definitely in a minority!
The incorrect manual pointed me at GitHub to get changes made, so here I am. My question is - how should I do this? I have looked all over the GitHub site but did not find a 'Corrections to Documentation' tab.

**Reply by tuves** [reply]
Hi @Electronic-Biker,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on this project, such as the [MicroPython](https://github.com/orgs/micropython/discussions). It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!

**Comment by Hex09AF** [comment]
Want to hear more, very nice "Best practices"

**Comment by Lyko22** [comment]
> ### Creating a username
> Welcome to GitHub! Thank you for starting your journey on the largest developer community in the world!
> 
> The first thing you need to do when using GitHub is create an account. Whether you're working in open source projects, working with friends on a school project, or just exploring a new tool that you overheard some people talking about at work, a GitHub account grants you basic access to the platform.
> 
> To create your account, you'll need to have 3 things:
> 
> * A unique username, or handle 🎉
> * An email address 📫
> * A password 🔒
> 
> ### Your handle
> At GitHub, your handle is super important. It is a public reference that can be traced back to all public (and sometimes private) things you do on GitHub. That can seem daunting, but it can also be fun! Handles at GitHub are an individual thing, no 2 are exactly the same, so take the time to decide on something that shows off the amazing person that you are. Selecting a handle can be a hard decision, thats where we can help!
> 
> #### Be yourself
> As stated above, handles are highly visible, so it only makes sense that they should show who you are as individual or who you want to be perceived as. Please keep in mind that a handle is also something that you should feel comfortable being public. Handles are often added to resumes so having a handle that is respectful may help recruiters feel comfortable about bringing in a candidate for a job interview.
> 
> Handles also area a great opportunity to show off more about yourself than your development. Maybe you have a love of a certain type of music, maybe you connect with a certain comic book character, or maybe you're an animal lover, etc. Keep in mind that trends sometimes pass, so creating a handle that is associated with a specific topic could age poorly.
> 
> Quick note: GitHub handles may only contain 39 alphanumeric characters and a hyphen (ex. `good-handle`). Hyphens can only be used once in a handle (ex. `not-allowed-handle`) and cannot be at the beginning or end of the handle (ex. `-notallowed-`). GitHub is a community for everyone, please consider reading [this page](https://docs.github.com/en/site-policy/acceptable-use-policies/github-hate-speech-and-discrimination) as well before creating a handle.
> 
> **Attention**: It is technically possible to change your GitHub handle, doing so will not update any references to your past handle. To learn more, review documentation found [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/changing-your-github-username).
> 
> At the end of the day, your GitHub handle will become your calling card when interacting with all things GitHub. Other GitHub users will use it as a reference when asking you questions, assigning you tasks, or looking at your public contributions. More info [here](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy).
> 
> ### Creating a repository
> Now that you have a handle, you can now create a repository. You can create repositories in a few places, but the easiest place to start is in your personal org. In the upper right corner of the GitHub.com, you will see a plus sign. Clicking on that will provide a prompt allowing you to create a new repository.
> 
> Repo creation screen should look something like this: ![repo-create-screenshot-1](https://user-images.githubusercontent.com/103224897/201693675-44049ae5-036d-40a2-b798-0337cc00aca3.png)
> 
> When creating a repository, the only required detail is the name of the repository. Repository names cannot have spaces, so any space you leave in your name will automatically be filled with dashes `-`. Optionally, you can include a description which can be used to help provide details about what is in the repository. For example, if you're working on a mobile app, you could name the repository "My-cool-app" and provide a description of "My mobile app written in React Native". This makes the repository much more approachable in the event that you want to share your cool app with the world!
> 
> #### Public or private repositories?
> When creating a repository you have the option of setting the repository as public or private. Public repositories are exactly as they sound, public! They are open for the world to see and can be viewed by anyone on GitHub. This is a good route if you're looking to share your work and potentially get public feedback and contributions. In contrast to public, private repositories are not visible to anyone other than you. With private repositories you also have the power to invite in other GitHub users to view and collaborate on your code.
> 
> The default choice when creating a repository is private, though it's not always the best choice. Say you're working on a new React Native app and you make a really slick menu bar. If your repository is public, then anyone who finds it would be able to use your code and improve their apps. By keeping your project public, anyone who finds your project could help you make improvements to your code through Issues and Pull Requests.
> 
> #### Initialize with a README.md?
> The final step to creating a repository is deciding whether to initiate it with a README. At GitHub, nearly all of our repositories have a README files and they serve a very important purpose. Creating your repository with a README is an excellent choice if you are starting from scratch and can help save time by avoiding the need to import an existing git project.
> 
> For those who are planning to import an existing git project, it is best to avoid selecting the 'initiate with README' checkbox. This will help you avoid conflicts in your git history when pushing the contents of your existing project into your new GitHub repository.
> 
> As stated above, at GitHub, the README file serves a very important purpose. The README(.md) file is extra special because it will be automatically shown as the first thing you see when viewing your GitHub repository from a web browser. Think of the README file as a guide to all of the things in your repository; it's the place where you get to share all of the important information about your repository and show others how to use your code. Visitors and contributors to your repository should be able to read through your project's README file and understand what your project does, who to contact for questions, how they can use it, and how to contribute.
> 
> README files are written in language called Markdown, which supports showing many varieties of information including images, links, GIFs, emojis, tables, and many others! While Markdown cheatsheets can be found all over the internet, GitHub does have some helpful docs that can assist you in getting you started [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).
> 
> To paint a picture, let's take a look at a README file for a very popular open source tool on GitHub, [Visual Studio Code](https://github.com/microsoft/vscode). The first thing you may notice in the README are some helpful 'badges'. These badges can be used to show all sorts of helpful information, in this example, VS Code has chosen to show issue counts and open bugs. Other popular badges include showing information about build status from a CI (Continuous Integration) tool, chat ops connection like Slack, or test coverage status. Next, as we move down the VS Code README, we see more information about the product, how to make contributions, how to provide feedback, and additional information about licensing, building, and a code of conduct. Not all of this information is needed in your README, but it can be very helpful to get an idea about what you'd like to share about your project.
> 
> The README files are a great starting point to your project, but they are far from the final step. You should expect your README file to change as your project changes, so when your React app grows to include some new features, you should expect your README to grow too!
> 
> ### Welcome to GitHub!
> That's all for now; welcome to GitHub. We are overjoyed that you have joined the largest developer community in the world. From amazing handle, to fresh code, we are excited build with you from here, together!

**Comment by Mian786a** [comment]
Please

**Comment by Luckythakurbit** [comment]
He'll o

**Comment by sudoshivesh** [comment]
Hello

**Comment by paule32** [comment]
what about code, with the different licenses ? - can the code mixed ?
great platform for hold up-to-date the source code.

**Comment by Meticu** [comment]
Hai

**Comment by Meticu** [comment]
![Uploading 20221113_150824.jpg…]()

**Comment by ghost** [comment]
> ![Uploading 20221113_150824.jpg…]()
> 
> 

👍

**Comment by ShukuraliProgrammer** [comment]
hi how are you

**Comment by wlanroot** [comment]
fine thanks how aBOUT U
    On Sunday, November 27, 2022 at 08:49:29 AM GMT+1, Shukurali ***@***.***> wrote:  
 
 


hi how are you

**Comment by eugenechekwubechukwu** [comment]
I want to delete my repositories

**Reply by dieguito1999** [reply]
don't you

**Reply by tuves** [reply]
Hi @eugenechekwubechukwu,

We're glad you're here in the GitHub Community! Please see instructions on [how to delete a repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/deleting-a-repository). For a more streamlined way to get answers to your questions, please drop them into the corresponding category. 

![Screen Shot 2022-12-05 at 9 49 56 AM](https://user-images.githubusercontent.com/49915749/205606615-34426bb7-824f-43d6-b617-646af8bb5f33.png)

For repository questions/feedback, please go [here](https://github.com/orgs/community/discussions/categories/repositories).

**Reply by revvv289** [reply]
Di mana saya bisa menemukan solusi saya bisa mendapat kan 16 digit nmor kartu ATM milik aaya sendiri

**Reply by revvv289** [reply]
Di mana saya bisa menemukan solusi saya bisa mendapat kan 16 digit nmor kartu ATM milik saya sendiri

**Comment by maghino98** [comment]
Thanks for de post, i´m new in GitHub and this information makes me feel safe jajaja.

**Comment by aceraver33** [comment]
> ### การสร้างชื่อผู้ใช้
> ยินดีต้อนรับสู่ GitHub! ขอขอบคุณที่เริ่มต้นการเดินทางในชุมชนนักพัฒนาซอฟต์แวร์ที่ใหญ่ที่สุดในโลก!
> 
> สิ่งแรกที่คุณต้องทำเมื่อใช้ GitHub คือสร้างบัญชี ไม่ว่าคุณจะทำงานในโครงการโอเพ่นซอร์ส ทำงานกับเพื่อนในโครงการโรงเรียน หรือแค่สำรวจเครื่องมือใหม่ที่คุณได้ยินบางคนพูดถึงในที่ทำงาน บัญชี GitHub ช่วยให้คุณเข้าถึงแพลตฟอร์มขั้นพื้นฐานได้
> 
> ในการสร้างบัญชีของคุณ คุณจะต้องมี 3 สิ่ง:
> 
> * ชื่อผู้ใช้หรือหมายเลขอ้างอิงที่ไม่ซ้ำกัน🎉
> * ที่อยู่อีเมล📫
> * รหัสผ่าน🔒
> 
> ### ที่จับของคุณ
> ที่ GitHub แฮนเดิลของคุณมีความสำคัญอย่างยิ่ง เป็นข้อมูลอ้างอิงสาธารณะที่สามารถตรวจสอบย้อนกลับไปยังกิจกรรมสาธารณะทั้งหมด (และบางครั้งเป็นส่วนตัว) ที่คุณทำบน GitHub อาจดูน่ากลัว แต่ก็สนุกได้เช่นกัน! แฮนเดิลที่ GitHub เป็นคนละเรื่องกัน ไม่มี 2 อันที่เหมือนกันทุกประการ ดังนั้นจงใช้เวลาในการตัดสินใจเลือกบางสิ่งที่แสดงให้เห็นว่าคุณเป็นคนที่น่าทึ่ง การเลือกด้ามจับอาจเป็นการตัดสินใจที่ยาก เราสามารถช่วยได้!
> 
> #### เป็นตัวของตัวเอง
> ตามที่ระบุไว้ข้างต้น แฮนเดิลสามารถมองเห็นได้อย่างชัดเจน ดังนั้นจึงเหมาะสมที่แฮนเดิลควรแสดงให้เห็นว่าคุณเป็นใครในฐานะปัจเจกบุคคลหรือคนที่คุณต้องการให้คนอื่นรับรู้ โปรดทราบว่าที่จับก็เป็นสิ่งที่คุณควรรู้สึกสบายใจเมื่อเปิดเผยต่อสาธารณะ มักจะมีการเพิ่มที่จับในเรซูเม่ ดังนั้นการมีที่จับที่ให้เกียรติอาจช่วยให้นายหน้ารู้สึกสบายใจในการนำผู้สมัครมาสัมภาษณ์งาน
> 
> Handles also area a great opportunity to show off more about yourself than your development. Maybe you have a love of a certain type of music, maybe you connect with a certain comic book character, or maybe you're an animal lover, etc. Keep in mind that trends sometimes pass, so creating a handle that is associated with a specific topic could age poorly.
> 
> Quick note: GitHub handles may only contain 39 alphanumeric characters and a hyphen (ex. `good-handle`). Hyphens can only be used once in a handle (ex. `not-allowed-handle`) and cannot be at the beginning or end of the handle (ex. `-notallowed-`). GitHub is a community for everyone, please consider reading [this page](https://docs.github.com/en/site-policy/acceptable-use-policies/github-hate-speech-and-discrimination) as well before creating a handle.
> 
> **Attention**: It is technically possible to change your GitHub handle, doing so will not update any references to your past handle. To learn more, review documentation found [here](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/changing-your-github-username).
> 
> At the end of the day, your GitHub handle will become your calling card when interacting with all things GitHub. Other GitHub users will use it as a reference when asking you questions, assigning you tasks, or looking at your public contributions. More info [here](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy).
> 
> ### Creating a repository
> Now that you have a handle, you can now create a repository. You can create repositories in a few places, but the easiest place to start is in your personal org. In the upper right corner of the GitHub.com, you will see a plus sign. Clicking on that will provide a prompt allowing you to create a new repository.
> 
> Repo creation screen should look something like this: ![repo-create-screenshot-1](https://user-images.githubusercontent.com/103224897/201693675-44049ae5-036d-40a2-b798-0337cc00aca3.png)
> 
> When creating a repository, the only required detail is the name of the repository. Repository names cannot have spaces, so any space you leave in your name will automatically be filled with dashes `-`. Optionally, you can include a description which can be used to help provide details about what is in the repository. For example, if you're working on a mobile app, you could name the repository "My-cool-app" and provide a description of "My mobile app written in React Native". This makes the repository much more approachable in the event that you want to share your cool app with the world!
> 
> #### Public or private repositories?
> When creating a repository you have the option of setting the repository as public or private. Public repositories are exactly as they sound, public! They are open for the world to see and can be viewed by anyone on GitHub. This is a good route if you're looking to share your work and potentially get public feedback and contributions. In contrast to public, private repositories are not visible to anyone other than you. With private repositories you also have the power to invite in other GitHub users to view and collaborate on your code.
> 
> The default choice when creating a repository is private, though it's not always the best choice. Say you're working on a new React Native app and you make a really slick menu bar. If your repository is public, then anyone who finds it would be able to use your code and improve their apps. By keeping your project public, anyone who finds your project could help you make improvements to your code through Issues and Pull Requests.
> 
> #### Initialize with a README.md?
> The final step to creating a repository is deciding whether to initiate it with a README. At GitHub, nearly all of our repositories have a README files and they serve a very important purpose. Creating your repository with a README is an excellent choice if you are starting from scratch and can help save time by avoiding the need to import an existing git project.
> 
> For those who are planning to import an existing git project, it is best to avoid selecting the 'initiate with README' checkbox. This will help you avoid conflicts in your git history when pushing the contents of your existing project into your new GitHub repository.
> 
> As stated above, at GitHub, the README file serves a very important purpose. The README(.md) file is extra special because it will be automatically shown as the first thing you see when viewing your GitHub repository from a web browser. Think of the README file as a guide to all of the things in your repository; it's the place where you get to share all of the important information about your repository and show others how to use your code. Visitors and contributors to your repository should be able to read through your project's README file and understand what your project does, who to contact for questions, how they can use it, and how to contribute.
> 
> README files are written in language called Markdown, which supports showing many varieties of information including images, links, GIFs, emojis, tables, and many others! While Markdown cheatsheets can be found all over the internet, GitHub does have some helpful docs that can assist you in getting you started [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).
> 
> To paint a picture, let's take a look at a README file for a very popular open source tool on GitHub, [Visual Studio Code](https://github.com/microsoft/vscode). The first thing you may notice in the README are some helpful 'badges'. These badges can be used to show all sorts of helpful information, in this example, VS Code has chosen to show issue counts and open bugs. Other popular badges include showing information about build status from a CI (Continuous Integration) tool, chat ops connection like Slack, or test coverage status. Next, as we move down the VS Code README, we see more information about the product, how to make contributions, how to provide feedback, and additional information about licensing, building, and a code of conduct. Not all of this information is needed in your README, but it can be very helpful to get an idea about what you'd like to share about your project.
> 
> The README files are a great starting point to your project, but they are far from the final step. You should expect your README file to change as your project changes, so when your React app grows to include some new features, you should expect your README to grow too!
> 
> ### Welcome to GitHub!
> That's all for now; welcome to GitHub. We are overjoyed that you have joined the largest developer community in the world. From amazing handle, to fresh code, we are excited build with you from here, together!

**Comment by ghost** [comment]
Plis kembalikan data saya yang lama

**Comment by Maxnegrao33** [comment]
I want to learn to work in Git, be new in this plataform.

**Comment by Hassanlasheenn** [comment]
That is so useful !

**Comment by SebaRomeroX** [comment]
Thanks this is really helpfull, i was kind of lost
